test/Elastic.Apm.Tests.Utilities/MockPayloadSender.cs

```diff
@@ -20,7 +20,10 @@ internal class MockPayloadSender : IPayloadSender
 	{
 		private readonly List<IError> _errors = new List<IError>();
 		private readonly List<Func<IError, IError>> _errorFilters = new List<Func<IError, IError>>();
-		private readonly object _lock = new object();
+		private readonly object _spanLock = new object();
```

Comment:
I saw tests failing with such logs - this specific test was a new one which I introduced in the PR. I wasn't able to repro this locally:

```
System.ArgumentException : Destination array was not long enough. Check the destination index, length, and the array's lower bounds.
Parameter name: destinationArray
Stacktrace
System.ArgumentException : Destination array was not long enough. Check the destination index, length, and the array's lower bounds.
Parameter name: destinationArray
Stack Trace:
   at System.Array.Copy(Array sourceArray, Int32 sourceIndex, Array destinationArray, Int32 destinationIndex, Int32 length, Boolean reliable)
   at System.Collections.Generic.List`1.CopyTo(T[] array, Int32 arrayIndex)
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at Elastic.Apm.Tests.BreakdownTests.TransactionWithSpans() in C:\Users\jenkins\workspace\net_apm-agent-dotnet-mbp_PR-1271\apm-agent-dotnet\test\Elastic.Apm.Tests\BreakdownTests.cs:line 62
   ```
   
  This exception isn't new - I already saw it on failing tests. I think the root cause of those failing tests is that the agent still puts an event into the `MockPayloadSender` while it makes the copy of the list. To address this I introduced locks in the `MockPayloadSender`.